src/fpPractice.test.js

```diff
@@ -0,0 +1,923 @@
+
+
+    //11th Question
+    //Find the second largest number in a given array first by using an imperative approach without using reduce. Use a forEach HOF to iterate over items and figure out the second largest item.
+    //Also give a solution using reduce method.
+
+    function findSecordLargest(arr) {
+        if (arr.length < 2) throw new Error("Array length less than 2")
+        let firstMax = arr[0]
+        let secondMax = arr[0]
+        arr.forEach(element => {
+            if (element > firstMax) {
+                secondMax = firstMax;
+                firstMax = element;
+            } else if (element > secondMax && element < firstMax) {
+                secondMax = element;
+            }
+        });
+
+        return secondMax;
+    }
+
+
+    expect(() => findSecordLargest([1])).toThrowError("Array length less than 2")
+    expect(findSecordLargest([1, 2])).toBe(1)
+    expect(findSecordLargest([1, 2, 1])).toBe(1)
+    expect(findSecordLargest([10, 11, 23, 44, 55])).toBe(44)
+    expect(findSecordLargest([10, 11, 12, 13, 14, 9, 11, 21])).toBe(14)
+    expect(findSecordLargest([10, 11, 22, 21])).toBe(21)
+
+
+    function findSecordLargestUsingReduce(arr) {
+        let newArr = [...arr]
+        if (arr.length < 2) throw new Error("Array length less than 2")
+        let largest = Math.max(...arr)
+        return newArr.reduce((secondLargest, currentValue) => {
+            return currentValue < largest && currentValue > secondLargest ? currentValue : secondLargest
+        }, -Infinity)
+    }
+
+    expect(() => findSecordLargestUsingReduce([1])).toThrowError("Array length less than 2")
+    expect(findSecordLargestUsingReduce([1, 2])).toBe(1)
+    expect(findSecordLargestUsingReduce([1, 2, 1])).toBe(1)
+    expect(findSecordLargestUsingReduce([10, 11, 23, 44, 55])).toBe(44)
+    expect(findSecordLargestUsingReduce([10, 11, 12, 13, 14, 9, 11, 21])).toBe(14)
+    expect(findSecordLargestUsingReduce([10, 11, 22, 21])).toBe(21)
+
+    //12th Question
+    //Implement a method called as   some  that takes the array as its first argument, and a predicate as its second argument.  Predicate is a function that takes an item in the array as its argument, and returns a boolean.  some function will return true if at-least one item in the array passes the predicate, otherwise will return false. Give an imperative solution, and then give a solution using reduce. 
+    //function some(items, predicate)
+
+    function some(arr, predicate) {
+        for (let num of arr) {
+            if (predicate(num)) return true
+        }
+        return false
+    }
+
+    expect(some([1, 2, 3, 4], (num) => num % 2 === 0)).toBeTruthy()
+    expect(some([1, 3, 5, 7], (num) => num % 2 === 0)).toBeFalsy()
+
+    function someUsingReduce(arr, predicate) {
+        let newArr = [...arr]
+        return newArr.reduce((result, currentValue) => {
+            if (predicate(currentValue)) return true
```

Comment:
there is a small issue with this implementation. This will fail, if your array contains all falsy items after a truthy item... For example the following test will fail:

```
expect(someUsingReduce([1, 3, 5, 7, 9, 11,12], (num) => num % 2 === 0)).toBeFalsy()

```